engine/api/src/main/scala/pl/touk/nussknacker/engine/api/dict/DictDefinition.scala

```diff
@@ -0,0 +1,41 @@
+package pl.touk.nussknacker.engine.api.dict
+
+import pl.touk.nussknacker.engine.api.typed.typing.{SingleTypingResult, Typed, TypedClass}
+
+/**
+ * It defines dictionary which will contain values of type `valueType`. Value will be created
+ * based on String `key`. `key` is presented as a `label` in UI.
+ *
+ * All definitions should be registered in `ProcessConfigCreator.expressionConfig().dictionaries`
+ * After registration, definitions will be available for `DictRegistry` and other services using it (like `DictQueryService`)
+ */
+trait DictDefinition extends Serializable {
+
+  def valueType(dictId: String): SingleTypingResult
+
+  // It should return value in declared type.
+  def value(key: String): Any
+
+}
+
+/**
+ * If this instance wil be used in global variables, will be typed to TypedDict.
+ */
+case class DictInstance(dictId: String, definition: DictDefinition) {
+
+  def valueType: SingleTypingResult = definition.valueType(dictId)
```

Comment:
Does DictInstance "contain" exactly one dictionary (map)? If so, it should not require dictId to return its type.

Reply:
`DictInstance` not need `dictId` to return its value type. Maybe you mean `DictDefinition`? It maybe confusing but our "instances" are objects that we put to global variables map. `DictIntance` refer to `DictDefinition` and `DictDefinition` is kind of abstract type. Concrete class is `DictDefinition` + dictId. Take a look at `TestProcessConfigCreator` to see how it used.

Reply:
ok, i think I get it